### PR TITLE
Bugfix/266 load path access type

### LINKED ISSUE #266: Bug: `pins.drivers.load_path` function uses built-in `type`
On lines 29 and 34, there seems to be a case of accessing a variable called `type` which is shadowing the built-in variable type; it is not defined elsewhere.

https://github.com/rstudio/pins-python/blob/641e5114f274091dc5c24327f345b79d9b052d04/pins/drivers.py#L1-L34

I suspect fixing this should just be adding `type = meta.type`.

To avoid this in the future, I suggest once `ruff` is added via #251 that we enable this rule:
https://docs.astral.sh/ruff/rules/builtin-variable-shadowing/

### PR DESCRIPTION
**Pull request by nathanjmcdougall**
To fix #266.

### CODE REVIEW

**Comment by nathanjmcdougall** [comment]
I realize that in my tests, I am mocking using a restricted Meta protocol similar to what is described in #12:
> Alternatively, there could be a stripped down version for Meta (or an interface), that defines the bare minimum required for data loading (i.e. type, filename).

This might be an opportunity to formally define a `Protocol` class and refactor the test suite accordingly.

File: pins/drivers.py [hunk 1]
```diff
@@ -25,12 +25,15 @@ def _assert_is_pandas_df(x, file_type: str) -> None:
 def load_path(meta, path_to_version):
     # Check that only a single file name was given
     fnames = [meta.file] if isinstance(meta.file, str) else meta.file
+
+    type = meta.type
```
Comment (isabelizimm) [inline pins/drivers.py:29]:
```suggestion
    _type = meta.type
```

I would slightly prefer this, so as not to be confusing with the builtin `type`.

File: pins/drivers.py [hunk 2]
```diff
@@ -25,12 +25,15 @@ def _assert_is_pandas_df(x, file_type: str) -> None:
...
+
     if len(fnames) > 1 and type in REQUIRES_SINGLE_FILE:
```
Comment (isabelizimm) [inline pins/drivers.py:31]:
```suggestion
    if len(fnames) > 1 and _type in REQUIRES_SINGLE_FILE:
```

File: pins/drivers.py [hunk 3]
```diff
@@ -25,12 +25,15 @@ def _assert_is_pandas_df(x, file_type: str) -> None:
...
         raise ValueError("Cannot load data when more than 1 file")
 
     # file path creation ------------------------------------------------------
 
-    if type == "table":  # noqa: E721 False Positive due to bug: https://github.com/rstudio/pins-python/issues/266
+    if type == "table":
```
Comment (isabelizimm) [inline pins/drivers.py:36]:
```suggestion
    if _type == "table":
```

**Review by isabelizimm — APPROVED** [review]
small naming nit, but otherwise looks good! Thank you for the tests as well 🙌

File: pins/drivers.py (hunk 1 above)
Comment (nathanjmcdougall) [reply pins/drivers.py:29]: Definitely.

**Pull request merged**